cmd/kaspawallet/daemon/server/address.go

```diff
@@ -6,35 +6,45 @@ import (
 	"github.com/kaspanet/kaspad/cmd/kaspawallet/daemon/pb"
 	"github.com/kaspanet/kaspad/cmd/kaspawallet/libkaspawallet"
 	"github.com/kaspanet/kaspad/util"
+	"github.com/pkg/errors"
 )
 
 func (s *server) changeAddress() (util.Address, error) {
+	s.keysFile.LastUsedInternalIndex++
+	err := s.keysFile.Save(true)
+	if err != nil {
+		return nil, err
+	}
+
 	walletAddr := &walletAddress{
-		index:         s.keysFile.LastUsedInternalIndex + 1,
+		index:         s.keysFile.LastUsedInternalIndex,
 		cosignerIndex: s.keysFile.CosignerIndex,
 		keyChain:      internalKeychain,
 	}
 	path := s.walletAddressPath(walletAddr)
-	s.keysFile.LastUsedInternalIndex++
 	return libkaspawallet.Address(s.params, s.keysFile.ExtendedPublicKeys, s.keysFile.MinimumSignatures, path, s.keysFile.ECDSA)
 }
 
 func (s *server) GetReceiveAddress(_ context.Context, request *pb.GetReceiveAddressRequest) (*pb.GetReceiveAddressResponse, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	err := s.validateIsSynced()
+	if !s.isSynced() {
+		return nil, errors.New("server is not synced")
+	}
+
+	s.keysFile.LastUsedExternalIndex++
+	err := s.keysFile.Save(true)
```

Comment:
Maybe it's a good idea to create setter/getter to LastUsedExternalIndex (and maybe any other such fields), that makes sure that any set it will also be persisted to disk?